src/hints.rs

```diff
@@ -284,43 +283,72 @@ impl<'a> From<&'a NotificationHint> for MessageItem {
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == ACTION_ICONS   => NotificationHint::ActionIcons(unwrap_message_bool(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == DESKTOP_ENTRY  => NotificationHint::DesktopEntry(unwrap_message_str(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == IMAGE_PATH     => NotificationHint::ImagePath(unwrap_message_str(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == RESIDENT       => NotificationHint::Resident(unwrap_message_bool(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == SOUND_FILE     => NotificationHint::SoundFile(unwrap_message_str(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == SOUND_NAME     => NotificationHint::SoundName(unwrap_message_str(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == SUPPRESS_SOUND => NotificationHint::SuppressSound(unwrap_message_bool(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == TRANSIENT      => NotificationHint::Transient(unwrap_message_bool(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == X              => NotificationHint::X(unwrap_message_int(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == Y              => NotificationHint::Y(unwrap_message_int(&**value)),
-            &MessageItem::DictEntry(ref key, ref value) if unwrap_message_str(&**key) == URGENCY        => NotificationHint::Urgency(
-                match unwrap_message_int(&**value){
-                    0 => NotificationUrgency::Low,
-                    2 => NotificationUrgency::Critical,
-                    _ => NotificationUrgency::Normal
-                }),
-            &MessageItem::DictEntry(ref key, ref value) => match try_unwrap_message_int(value) {
-                    Some(num) => NotificationHint::CustomInt(unwrap_message_str(&**key), num),
-                    None => NotificationHint::Custom(unwrap_message_str(&**key), unwrap_message_str(&**value))
-            }
-            other => {
-                println!("Invalid {:#?} ", other);
-                NotificationHint::Invalid
-            }
+impl From<&MessageItem> for NotificationHint {
+
+    fn from(_: &MessageItem) -> Self {
+
+        // This is kept for backwards-compatibility. The ability to transform
+        // a single MessageItem into a NotificationHint was removed in dbus 0.7
+        // when DictEntry was removed
+        Self::Invalid
+    }
+}
+
+#[deprecated(note = "Prefer the DBus Arg and RefArg APIs")]
```

Comment:
what does this mean? is there a nicer way to implement this?
this is not really something a user of this lib should be doing anyway

Reply:
I think it's just for the server implementation, and that's bit of an ongoing pain too, its only there so I can test what I send ATM